vowpalwabbit/cb_adf.cc

```diff
@@ -188,6 +188,8 @@ void output_example(vw& all, cb_adf& c, example& ec, v_array<example*>* ec_seq)
   float loss = 0.;
 
   uint32_t action = ec.pred.a_s[0].action;
+  ec.pred.a_s.delete_v();
```

Comment:
It's not.  There are two issues:
(1) VW generally follows a strategy of allocation at the beginning of the computation and deallocation at the end of the computation. This violates that, and a deallocation here implies that you must insert an allocation somewhere else.  
(2) pred is a union so extra care must be taken.  Here, you are calling delete_v on a piece of memory interpreted as a v_array<action_score>, then using the same memory as a uint32_t.  This will cause problems...